tritond/main_test.go

```diff
@@ -40,20 +41,20 @@ type consumer struct {
 }
 
 func (c *consumer) Start() error {
-	var err error
-	c.socket, err = zmq4.NewSocket(zmq4.PULL)
-	if err != nil {
-		return err
-	}
-
-	if err = c.socket.Bind("tcp://127.0.0.1:3515"); err != nil {
-		return err
-	}
-
 	go func() {
+		var err error
+		c.socket, err = zmq4.NewSocket(zmq4.PULL)
+		if err != nil {
+			panic(err)
+		}
+
+		if err = c.socket.Bind("tcp://127.0.0.1:3515"); err != nil {
+			panic(err)
```

Comment:
This error is much harder to deal with than the previous serial version.

Reply:
why not just return the err on a channel (or put it in an error slice) and wait for all go routines to finish then check the error and return it?